sycl/include/CL/sycl/handler.hpp

```diff
@@ -198,6 +198,42 @@ checkValueRange(const T &V) {
 #endif
 }
 
+template <typename TransformedArgType, int Dims, typename KernelType>
+struct RoundedRangeKernel {
+public:
+  RoundedRangeKernel(range<Dims> NumWorkItems, KernelType KernelFunc)
+      : NumWorkItems(NumWorkItems), KernelFunc(KernelFunc) {}
+
+  void operator()(TransformedArgType Arg) const {
+    if (Arg[0] >= NumWorkItems[0])
+      return;
+    Arg.set_allowed_range(NumWorkItems);
+    KernelFunc(Arg);
+  }
+
+private:
+  range<Dims> NumWorkItems;
+  KernelType KernelFunc;
+};
+
+template <typename TransformedArgType, int Dims, typename KernelType>
+struct RoundedRangeKernelWithKH {
+public:
+  RoundedRangeKernelWithKH(range<Dims> NumWorkItems, KernelType KernelFunc)
+      : NumWorkItems(NumWorkItems), KernelFunc(KernelFunc) {}
+
+  void operator()(TransformedArgType Arg, kernel_handler KH) const {
```

Comment:
Would it be possible to have this operator()(TransformedArgType Arg, kernel_handler KH) as overload in RoundedRangeKernel ? I.e. not create an additional class RoundedRangeKernelWithKH ?

Reply:
I did that to preserve the previous semantics. You only can call one of them but not both, which is controlled by the following code in `handler.hpp`:
```
if constexpr (detail::isKernelLambdaCallableWithKernelHandler<...>()) {
```